src/Neo/Wallets/NEP6/NEP6Wallet.cs

```diff
@@ -71,6 +72,7 @@ public NEP6Wallet(string path, string password, ProtocolSettings settings, strin
                 accounts = new Dictionary<UInt160, NEP6Account>();
                 extra = JToken.Null;
             }
+            Thread.MemoryBarrier();
```

Comment:
If the object is not yet constructed, how it can be dangerous in multi-thread?

Reply:
Because there are some use cases of `Wallet` that wallet is a field of a class(for example, `RpcServer`) and it is loaded on needed. i.e.  Wallet field is null before `Wallet` load.

A wallet load operation will assign a `Wallet` instance to this field, so there is such a case that can happen:
Thread 1 load a wallet and assign wallet instance to wallet field, but this instance is partial constructed  because of Out-of-Order execution of CPU;

Thread 2 can see this partial constructed instance.


This is a classic concurrency problem.

In Java, this can be solved by declaring a field as `final`;

In Linux kernel, `WRITE_ONCE` and `READ_ONCE` is often used.

In C#, this can be solved by declaring a field as `volatile`, but this fix uses memory barrier.

Reply:
The variable is not assigned yet, only after is constructed, so you need to do this after this assignation, not before

Reply:
> The variable is not assigned yet, only after is constructed, so you need to do this after this assignation, not before

In another thread‘ view, a wallet may assigned to a Wallet field, then this wallet instance’s fields are assigned to this instance because of out of order execution

Reply:
> The variable is not assigned yet, only after is constructed, so you need to do this after this assignation, not before

In another thread‘ view, a wallet may assigned to a Wallet field, then this wallet instance’s fields are assigned to this instance because of out of order execution